### PR TITLE
Update dataflow python to use external containers

### PR DESCRIPTION
**Pull request by damccorm**
This updates python to use containers created as part of the Beam snapshot process instead of containers built by Dataflow.

These containers are initially published to https://pantheon.corp.google.com/gcr/images/apache-beam-testing/global/beam_portability/ and then mirrored to https://pantheon.corp.google.com/gcr/images/cloud-dataflow/global/v1beta3/ (for example, for 3.8 containers are mirrored to - https://pantheon.corp.google.com/gcr/images/cloud-dataflow/global/v1beta3/beam_python3.8_sdk)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesContainer

**Comment by damccorm** [comment]
R: @tvalentyn

**Review by tvalentyn — APPROVED** [review]

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py [hunk 1]
```diff
@@ -1207,19 +1207,21 @@ def get_container_image_from_options(pipeline_options):
...
-  image_name = '{repository}/python{version_suffix}{fnapi_suffix}'.format(
-      repository=names.DATAFLOW_CONTAINER_IMAGE_REPOSITORY,
-      version_suffix=version_suffix,
-      fnapi_suffix=fnapi_suffix)
+  container_repo = names.DATAFLOW_CONTAINER_IMAGE_REPOSITORY
+  image_name = '{repository}/python{version_suffix}'.format(
+      repository=container_repo, version_suffix=version_suffix)
+
+  if is_runner_v2:
+    version_suffix = '%s.%s' % (sys.version_info[0:2])
```
[the remark this answers is not in the document]
Comment (damccorm) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: We actually need this, it inserts a . while the last one does not.

With that said, I don't think this is helpful (the difference is too subtle), so I removed version_suffix entirely and replaced it with major/minor

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesContainer

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesContainer

**Pull request merged**